File: repos_estudiantes_ALVAREZ_primera_entrega/CARDON_PAZ_LUIS_ABDEL/INTRODUCCION_POO/EJERCICIO_2/Bus.py

class Bus:
    pasaje = 1.50

    def __init__(self, capacidad):
        self.capacidad = capacidad
        self.pasajeros = 0
        self.ganancia = 0.0

    def subir_pasajeros(self, cantidad):
        if cantidad <= 0:
            print("No subio ningun pasajero.")
            return
        elif cantidad + self.pasajeros > self.capacidad:
            print("No hay espacio suficiente en el bus.")
            return
        else:
            self.pasajeros += cantidad
            print(f"Subieron: {cantidad} pasajeros")

    def cobrar_pasaje(self):
        total = self.pasajeros * Bus.pasaje
        self.ganancia += total
        return self.ganancia

    def asientos_disponibles(self):
        return self.capacidad - self.pasajeros

    def __str__(self):
        return f"Capacidad: {self.capacidad}, Pasajeros: {self.pasajeros}, Ganancia: {self.ganancia} Bs"



bus = Bus(50)
print("prueba 1")
bus.subir_pasajeros(17)
print("Asientos disponibles:", bus.asientos_disponibles())
print("Ganancia total:", bus.cobrar_pasaje(), "Bs")
print(str(bus) + "\n")
print("prueba 2")
bus.subir_pasajeros(52)

File: repos_estudiantes_ALVAREZ_primera_entrega/CARDON_PAZ_LUIS_ABDEL/INTRODUCCION_POO/EJERCICIO_2/Bus.java
package INTRODUCCION_POO.EJERCICIO_2;

public class Bus{
    public int capacidad;
    public int pasajeros;
    private double ganancia;
    private static final double pasaje= 1.50;
    

    public Bus(int capacidad){
        this.capacidad = capacidad;
        this.pasajeros = 0;
        this.ganancia = 0.0;
    }

    public void subirPasajeros(int cantidad){
        if(cantidad <= 0){
            System.out.println("No subio ningun pasajero.");
            return;
        }
        else if (cantidad + this.pasajeros > capacidad){
            System.out.println("No hay espacio suficiente en el bus.");
            return;
        }else {
            this.pasajeros = this.pasajeros + cantidad;
            System.out.println("Subieron: " + cantidad + " pasajerons");
        }
    }

    public double cobrarPasaje(){
        double total = this.pasajeros * pasaje;
        this.ganancia = this.ganancia + total;
        double gananciaTotal = this.ganancia;
        return gananciaTotal;
    }

    public int asientosDisponibles(){
        int disponibles = this.capacidad - this.pasajeros;
        return disponibles;
    }

    public String toString(){
        return "Capacidad: " + capacidad + ", Pasajeros: " + pasajeros + ", Ganancia: " + ganancia + " Bs";
    }
}
